components/dashboard/sideMenu/drawer.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { ReactNode } from "react";
+import { CSSObject } from "@mui/material/styles";
+import MuiDrawer from "@mui/material/Drawer";
+import { theme } from "@/src/utils/theme";
+
+const drawerWidth = 270;
+
+interface Props {
+    children: ReactNode;
+    open: boolean;
+    setOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const openedMixin = (): CSSObject => ({
+    width: drawerWidth,
+    transition: theme.transitions.create("width", {
+        easing: theme.transitions.easing.sharp,
+        duration: theme.transitions.duration.enteringScreen,
+    }),
+    overflowX: "hidden",
+});
+
+const closedMixin = (): CSSObject => ({
+    transition: theme.transitions.create("width", {
+        easing: theme.transitions.easing.sharp,
+        duration: theme.transitions.duration.leavingScreen,
+    }),
+    overflowX: "hidden",
+    width: `calc(${theme.spacing(7)} + 1px)`,
+    [theme.breakpoints.up("sm")]: {
+        width: `calc(${theme.spacing(8)} + 1px)`,
+    },
+});
+
+export const Drawer: React.FC<Props> = ({ open, setOpen, children }) => {
+    const handleDrawerOpen = () => {
+        setOpen(true);
+    };
+
+    const handleDrawerClose = () => {
+        setOpen(false);
+    };
+
+    return (
+        <MuiDrawer
+            onMouseEnter={handleDrawerOpen}
```

Comment:
Opravdu potřebuješ ty dvě funkce?
```suggestion
            onMouseEnter={setOpen(boolean)}
```

Reply:
To se vlastně spustí při renderu. V tom případě
```js
() => setOpen(boolean)
```